crates/forge_loader/src/manifest.rs

```diff
@@ -59,23 +81,159 @@ enum Interval {
     Week,
 }
 
+// Maps to Scheduled Trigger under Common Modules
 #[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
 struct ScheduledTrigger<'a> {
     #[serde(flatten, borrow)]
     raw: RawTrigger<'a>,
     interval: Interval,
 }
 
+// Maps to Web Trigger under Common Modules
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+struct RawTrigger<'a> {
+    key: &'a str,
+    function: &'a str,
+}
+
+// maps to Trigger under Common Modules
 #[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
-struct WorkflowPostFunction<'a> {
+struct EventTrigger<'a> {
     #[serde(flatten, borrow)]
     raw: RawTrigger<'a>,
+    #[serde(borrow)]
+    events: Vec<&'a str>,
 }
 
+// Compass Modules
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct CompassAdminPage<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct ComponentPage<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct DataProvider<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    callback: JustFunc<'a>,
```

Comment:
```suggestion
    callback: Option<JustFunc<'a>>,
```
This field is optional according to the [docs](https://developer.atlassian.com/platform/forge/manifest-reference/modules/compass-data-provider/#properties).